test/fw/ptl/lib/ptl_wrappers.py

```diff
@@ -1857,6 +1858,293 @@ def submit(self, obj, script=None, extend=None, submit_dir=None,
 
         return objid
 
+    def submit_resv(self, offset, duration, select='1:ncpus=1', rrule='',
+                    times=False, conf=None):
+        """
+        Helper function to submit an advance/a standing reservation.
+        :param int offset: Time in seconds from time this is called to set the
+                       advance reservation's start time.
+        :param int duration: Duration in seconds of advance reservation
+        :param str select: Select statement for reservation placement.
+                           Default: "1:ncpus=1"
+        :param str rrule: Recurrence rule.  Default is an empty string.
+        :param boolean times: If true, return a tuple of reservation id, start
+                              time and end time of created reservation.
+                              Otherwise return just the reservation id.
+                              Default: False
+        :param conf: Configuration for test case for PBS_TZID information.
+        :return The reservation id if times is false.  Otherwise a tuple of
+                reservation id, start time and end time of the reservation.
+
+        """
+        start_time = int(time.time()) + offset
+        end_time = start_time + duration
+
+        attrs = {
+            'reserve_start': start_time,
+            'reserve_end': end_time,
+            'Resource_List.select': select
+        }
+
+        if rrule:
+            if conf is None:
+                self.logger.info('conf not set. Falling back to Asia/Kolkata')
+                tzone = 'Asia/Kolkata'
+            elif 'PBS_TZID' in conf:
+                tzone = conf['PBS_TZID']
+            elif 'PBS_TZID' in os.environ:
+                tzone = os.environ['PBS_TZID']
+            else:
+                self.logger.info('Missing timezone, using Asia/Kolkata')
+                tzone = 'Asia/Kolkata'
+            attrs[ATTR_resv_rrule] = rrule
+            attrs[ATTR_resv_timezone] = tzone
+
+        rid = self.submit(Reservation(TEST_USER, attrs))
+
+        if times:
+            return rid, start_time, end_time
+        return rid
```

Comment:
There are existing tests that were using this function as-defined internal to their files, notably the resv_end_hook tests, and returning the start and end times became necessary when working on the modify_resv tests (begin, confirm, and end did not need this at all).

I can go through and change the calls that are using this, though.

Reply:
Since you are adding this to PTL itself, I'd rather us do it right.  If you don't mind changing the other tests, that'd be great.  It shouldn't be hard to add '[0]' at the end of all the function calls which don't need the start or end times.

Reply:
Because of this change, all of the PTL test changes should be in 00d1d03.  I'll note that they should have been addressed in that revision.